File: shared-backend/models/CarModel.js
const mongoose = require('mongoose');

const carModelSchema = new mongoose.Schema({
  brandId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'CarBrand',
    required: true
  },
  brandName: {
    type: String,
    required: true,
    trim: true
  },
  name: {
    type: String,
    required: true,
    trim: true
  },
  yearStart: {
    type: Number,
    default: null
  },
  yearEnd: {
    type: Number,
    default: null
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

// Index for efficient queries
carModelSchema.index({ brandId: 1, isActive: 1 });
carModelSchema.index({ brandName: 1, name: 1 });

module.exports = mongoose.model('CarModel', carModelSchema);

File: clutch-app-android/app/src/main/java/com/clutch/app/data/model/CarModel.kt
package com.clutch.app.data.model

import com.google.gson.annotations.SerializedName

data class CarModel(
    @SerializedName("_id")
    val id: String,
    @SerializedName("brandId")
    val brandId: String,
    @SerializedName("brandName")
    val brandName: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("yearStart")
    val yearStart: Int? = null,
    @SerializedName("yearEnd")
    val yearEnd: Int? = null,
    @SerializedName("isActive")
    val isActive: Boolean = true
)
